### PR TITLE
(FACT-2220) Add Aix disks fact

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

File: lib/facter/resolvers/aix/disks.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+              size_bytes: size_bytes,
+              size: Facter::FactsUtils::UnitConverter.bytes_to_human_readable(size_bytes)
+            }
+          end
+
+          def extract_info(lsl_content)
+            lsl_content = lsl_content.strip.split("\n").map do |line|
+              next unless /PP SIZE:|TOTAL PPs:|FREE PPs:|PV STATE:/ =~ line
+
+              line.split(/:|\s\s/)
```
Comment (IrimieBogdan) [inline lib/facter/resolvers/aix/disks.rb:56]: We could improve the regex so it does not generate so many empty value 
```
line.split(/:\s*|\s{2,}/)
```

File: lib/facter/resolvers/aix/disks.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            end
+
+            lsl_content.flatten!.select! { |elem| elem }.map! { |elem| elem.delete("\s") }.reject!(&:empty?)
```
Comment (IrimieBogdan) [inline lib/facter/resolvers/aix/disks.rb:59]: If we use the improve regex at line 56, we can simplify this to 
```
lsl_content.flatten!.select! { |elem| elem }
```

File: lib/facter/resolvers/aix/disks.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+            size_bytes = compute_size(info_hash)
+
+            {
+              size_bytes: size_bytes,
+              size: Facter::FactsUtils::UnitConverter.bytes_to_human_readable(size_bytes)
+            }
+          end
+
+          def extract_info(lsl_content)
```
Comment (IrimieBogdan) [inline lib/facter/resolvers/aix/disks.rb:52]: `lsl_content` should be `lsp_content`

File: lib/facter/resolvers/aix/disks.rb (hunk 1 above)
Comment (oanatmaria) [reply lib/facter/resolvers/aix/disks.rb]: done

File: lib/facter/resolvers/aix/disks.rb (hunk 2 above)
Comment (oanatmaria) [reply lib/facter/resolvers/aix/disks.rb]: done

File: lib/facter/resolvers/aix/disks.rb (hunk 3 above)
Comment (oanatmaria) [reply lib/facter/resolvers/aix/disks.rb]: done

File: lib/facter/resolvers/aix/disks.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+module Facter
+  module Resolvers
+    module Aix
+      class Disks < BaseResolver
+        @semaphore = Mutex.new
+        @fact_list ||= {}
+        class << self
+          private
+
+          MEGABYTES_EXPONENT = 1024**2
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/facter/resolvers/aix/disks.rb]: Done

File: lib/facter/resolvers/aix/disks.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          def execute_lspv(fact_name)
+            @fact_list[:disks] = {}
+
+            result = Facter::Core::Execution.execute('lspv', logger: log)
+
+            return if result.empty?
+
+            result.each_line do |line|
+              disk_name = line.split(' ')[0].strip
+              info = find_size(disk_name)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/facter/resolvers/aix/disks.rb:25]: Done

File: lib/facter/resolvers/aix/disks.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+              line.split(/:|\s\s/)
+            end
+
+            lsl_content.flatten!.select! { |elem| elem }.map! { |elem| elem.delete("\s") }.reject!(&:empty?)
+
+            Hash[*lsl_content]
+          end
+
+          def compute_size(info_hash)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/facter/resolvers/aix/disks.rb]: Done

File: lib/facter/resolvers/aix/disks.rb [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            size_bytes = compute_size(info_hash)
+
+            {
+              size_bytes: size_bytes,
+              size: Facter::FactsUtils::UnitConverter.bytes_to_human_readable(size_bytes)
+            }
+          end
+
+          def extract_info(lsl_content)
+            lsl_content = lsl_content.strip.split("\n").map do |line|
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/facter/resolvers/aix/disks.rb]: Done

File: spec/facter/resolvers/aix/disks_spec.rb [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    end
+
+    context 'when second lspv call fails' do
+      before do
+        allow(Facter::Core::Execution).to receive(:execute).with('lspv hdisk0', logger: logger_spy)
+                                                           .and_return('')
+      end
+
+      it 'returns disks informations' do
+        expect(resolver.resolve(:disks)).to eq({})
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/facter/resolvers/aix/disks_spec.rb:49]: there are two calls of lspv, this test is for the scenario when the second one returns empty (there are 2 imbricated contexts here) and the one at line 22 is the scenario when the first one returns empty

**Comment by oanatmaria** [comment]
jenkins please test this on aix61-POWERa

**Comment by oanatmaria** [comment]
jenkins please test this on aix61-POWERa

**Pull request merged**